### PR TITLE
(BOLT-1284) Add skeleton bolt project directory

### PR DESCRIPTION
**Pull request by donoghuc**
This commit adds a skeleton bolt project directory to enable new users to quickly get started with bolt.

### CODE REVIEW

File: bolt.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# puppetdb:
+#   server_urls: ["https://foo.example.com:8081"]
+#   cacert: /path/to/ca.pem
+#   token: ~/.puppetlabs/token
+# pcp:
+#   service-url: "https://foo.example.com:8143"
+#   cacert: /path/to/ca.pem
+#   task-environment: production
+# ssh:
```
Comment (donoghuc) [inline bolt.yaml]: For most skeleton files I added some commented out examples to stub out an idea for making a working example project that actually uses all the example files. Is that worth investing the time in to? Or do we want to limit this to simply a directory structure?

File: README.md [hunk 2]
```diff
@@ -1 +1,63 @@
...
+The default location for hiera configuration. [Hiera configuration reference](https://puppet.com/docs/puppet/latest/hiera_config_yaml_5.html)
+
+### inventory.yaml
+The `inventory.yaml` file contains information about targets. You can group targets and define connection configuration. [inventory.yaml reference](https://puppet.com/docs/bolt/latest/inventory_file.html)
+
+### Puppetfile
+The `Puppetfile` is used to install modules from the [Puppet Forge](https://forge.puppet.com/) or from a Git repository. Modules specified in the Puppetfile will be downloaded and installed into the `modules` directory. [Puppetfile reference](https://puppet.com/docs/bolt/latest/installing_tasks_from_the_forge.html#task-8928)
+
+### site-modules
+The `site-modules` directory containts the project specific `Bolt` modules. In this example there is a single module called `project`.
```
Comment (nicklewis) [inline README.md]:
```suggestion
The `site-modules` directory contains the project specific `Bolt` modules. In this example there is a single module called `project`.
```

File: bolt.yaml (hunk 1 above)
Comment (nicklewis) [reply bolt.yaml]: I think that if we're going to have example content in these files, we should include some comments at least 
 explaining what the sections are for.

File: bolt.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# puppetdb:
+#   server_urls: ["https://foo.example.com:8081"]
+#   cacert: /path/to/ca.pem
+#   token: ~/.puppetlabs/token
+# pcp:
+#   service-url: "https://foo.example.com:8143"
+#   cacert: /path/to/ca.pem
+#   task-environment: production
```
Comment (nicklewis) [inline bolt.yaml]: We should probably omit `task-environment`, since that's quite an advanced use case

File: bolt.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+# ssh:
+#   host-key-check: false
+#   private-key: ~/.ssh/bolt_id
+#   user: foo 
+#   interpreters:
```
Comment (nicklewis) [inline bolt.yaml]: `interpreters` is a complex topic that we maybe shouldn't cover in this file

File: bolt.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+#      rb: /home/foo/.rbenv/versions/2.5.1/bin/ruby
+# log:
+#   console:
```
Comment (nicklewis) [inline bolt.yaml]: Configuring the console log level is a strange use case given BOLT-901, so we might want to omit that as well

File: data/common.yaml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+# ---
+# hiera_data: "special data"
```
Comment (nicklewis) [inline data/common.yaml]: This key name is kind of meta and makes it read like this is a config file for hiera. We should use something more generic, and probably include a comment explaining the structure of the file

File: site-modules/project/manifests/init.pp [hunk 7]
```diff
@@ -0,0 +1,3 @@
+# class project(){
```
Comment (nicklewis) [inline site-modules/project/manifests/init.pp]:
```suggestion
# class project() {
```

File: site-modules/project/plans/init.pp [hunk 8]
```diff
@@ -0,0 +1,3 @@
+# plan project(){
```
Comment (nicklewis) [inline site-modules/project/plans/init.pp]:
```suggestion
# plan project() {
```

File: site-modules/project/tasks/init.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
+// {
```
Comment (nicklewis) [inline site-modules/project/tasks/init.json]: Could we start them with a real task here? Maybe a simple "echo" or something.

Some of the features demonstrated in this sample metadata are a bit advanced, like `implementations`. Also, comments aren't supported here, which means this file will likely cause an error if you tried to run the task.

File: Puppetfile [hunk 10]
```diff
@@ -0,0 +1,4 @@
+#forge 'http://forge.puppetlabs.com'
+#
+#mod 'profiles', local: true
```
Comment (nicklewis) [inline Puppetfile]: I don't think you really need to use `local` in a Boltdir, since those should be in `site-modules` anyhow

File: README.md [hunk 11]
```diff
@@ -1 +1,63 @@
...
+* Executes commands on remote *nix and Windows systems*.
+* Distributes and execute scripts, such as Bash, PowerShell, Python.
+* Scales to more than 1000 concurrent connections.
+* Supports industry standard protocols (SSH/SCP, WinRM/PSRP) and authentication methods (password, publickey).
+
+> Installing bolt from a gem is not recommended since core modules will not be available. Please [install bolt](https://puppet.com/docs/bolt/latest/bolt_installing.md) as a package
+
+This repository is an example skeleton Bolt Project Directory or `Boltdir`. A `Boltdir` contains all of the configuration, code, and data loaded by Bolt. Read more about the project directory structure at the following [link.](https://puppet.com/docs/bolt/latest/bolt_project_directories.html#project-directories)
+
+## Directory structure
```
Comment (lucywyman) [inline README.md:14]: I think this should just be a link to the docs page, that way there's only 1 source of truth.

File: bolt.yaml (hunk 1 above)
Comment (lucywyman) [reply bolt.yaml]: If the main purpose for this is to have something folks can clone and get started with at contributor summit I don't think content is necessary. People will be familiar with the file names and can refer to docs, and we'll have lots of Bolt knowledgeable folks on hand to help. I think just having empty files is fine, and we can add content if we want to reuse this for something else.

**Review by lucywyman — CHANGES_REQUESTED** [review]
2 comments, but should be quick to address! Thanks for doin' this Cas - I do think it'll make the getting started stuff at CS much easier.

File: Puppetfile [hunk 12]
```diff
@@ -0,0 +1,8 @@
+## This is an example puppetfile. 
+
+## The commented out section shows the URL of the Puppet forge to use.
```
Comment (adreyer) [inline Puppetfile]: We should not include the "forge" directive. It's more or less deprecated and wrong.

File: README.md (hunk 11 above)
Comment (donoghuc) [reply README.md:14]: I think it is important to leave this as is. I link to the docs about choosing an appropriate directory structure above. In terms of a "source of truth" this shows the directory structure of this repository. So the only change would be if we added or removed a file from this project.

**Comment by donoghuc** [comment]
Based on conversation in slack, I updated to only include a short description of each file and a link to docs as a comment in each file.

**Pull request merged**